src/app/rpmostree-builtin-kargs.c

```diff
@@ -0,0 +1,203 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*-
+ *
+ * Copyright (C) 2017 Red Hat Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; either version 2 of the licence or (at
+ * your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General
+ * Public License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place, Suite 330,
+ * Boston, MA 02111-1307, USA.
+ */
+
+#include "config.h"
+
+#include "rpmostree-ex-builtins.h"
+#include "rpmostree-libbuiltin.h"
+#include "rpmostree-dbus-helpers.h"
+
+#include <libglnx.h>
+
+/* TODO uncomment this when editor functionality is done
+ *  static gboolean opt_editor;
+ */
+static gboolean opt_import_proc_cmdline;
+static gboolean opt_reboot;
+static char **opt_kernel_delete_strings;
+static char **opt_kernel_append_strings;
+static char **opt_kernel_replace_strings;
+static char  *opt_osname;
+static char  *opt_deployid;
+
+static GOptionEntry option_entries[] = {
+  { "os", 0, 0, G_OPTION_ARG_STRING, &opt_osname, "Operation on provided OSNAME", "OSNAME" },
+  { "deployid", 0, 0, G_OPTION_ARG_STRING, &opt_deployid, "Modify the kernel args from a specific deployment based on id. Id is in the form of 'osname-checksum.deployserialnum' ", "DEPLOYID"},
```

Comment:
I see this is used in the DBus API too.  Hmm.  I'm not sure.  I think my tentative vote here is to use an index in the API too but I could definitely be convinced to try to standardize this "deployid" too.

Reply:
> I think my tentative vote here is to use an index in the API too

Sure, index here makes more sense. I used the deploy id implementation just simply because this was also used in the old DBus methods :p. Will do a change.